### ISSUE TITLE
Motoko 0.7.3 (source qwdq7q2n-npg4rjz7-7s76jv69-kbwjwklg)

### ISSUE THREAD

**Issue by tomijaga** [dfinity/motoko#3539]

**Comment by crusso** [comment]
Thanks for reporting this.

If possible, can you provide the code (or a cut down version of it) that produces the bug, as well as the moc compiler arguments used to compile the code. I'm particulay interested in whether you are interpreting or compiling the code, and for which target.

**Comment by tomijaga** [comment]
I think this is the block of code that caused the issue: https://github.com/NatLabs/serde/blob/main/tests/Json.Test.mo#L32

I used this command to compile the code: https://github.com/NatLabs/serde/blob/main/makefile#L6
I don't know what you mean by target, but I'm using an m1 mac with Rosetta enabled.

**Comment by tomijaga** [comment]
@ggreif 
https://github.com/dfinity/motoko/issues/3540#issuecomment-1300015568
> This shouldn't happen. Can you describe why the problem came and then went away?

After looking at it again. I think the issue is I am trying to compile more that one into an object file.
It works when I remove the `tests/*Test.mo` command from https://github.com/NatLabs/serde/blob/main/makefile#L6 and specify a single file.
Is there a command for compiling multiple files?

**Comment by ggreif** [comment]
Heh, this must be some old vestige in the `moc` compiler. Even I didn't know that you can use it to compile various files into one `.wasm`:
``` make
test1:
	$(shell vessel bin)/moc $(shell vessel sources) -wasi-system-api -o Test.wasm tests/*Test.mo && wasmtime test.wasm && rm -f Test.wasm
```

**Comment by crusso** [comment]
Thanks for the repo! This is indeed a left-over feature from the days when moc was just an interpreter. Maybe we should just refuse more gracefully to handle this when actually compiling...

**Comment by ggreif** [comment]
Yes, we should part from the `gcc` idea to compile several files at the same time. The list of sources should have at most one entry.
